src/components/changes-only-diff.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import DiffItem from './diff-item';
+import List from './list';
+
+/**
+ * Display a plaintext diff with additions and removals inline.
+ *
+ * @class ChangesOnlyDiff
+ * @extends {React.Component}
+ * @param {Object} props
+ * @param {ChangeDiff} props.diff
+ * @param {string} props.className
+ */
+export default class ChangesOnlyDiff extends React.Component {
+  render () {
+    if (!this.props) {
+      return null;
+    }
+
+    const contextDiff = this.props.diff.content.diff.map(getContextualDiff);
+
+    return (
+      <List
+        data={contextDiff}
+        component={DiffItem}
+        className={this.props.className}
+      />
+    );
+  }
+}
+
+/**
+   * Truncating function to process changes-only diffs.
+   * If `currentValue` is an insertion or deletion we return it as is.
+   * If it's not, we truncate it depending on if changes occured in entries before or after it.
+   * @param {array|Object} currentValue
+   * @param {number} index
+   * @param {diff[]} diffs
+   * @returns {[number, string]}
+   */
+function getContextualDiff (currentValue, index, diffs) {
+  let itemType;
+  let itemText;
+  // QUESTION: This logic comes from diff-item.jsx.
+  // Different services return slightly different json results,
+  // with what would be corresponding values switched and in different format???
+  // `html_text`: array of array values [number, string]
+  // `source`: array of objects {Text<string>, Type<number>}
+  // Should this be normalized?
+  if (currentValue.constructor === Array) {
+    [itemType, itemText] = currentValue;
+  }
+  else {
+    itemType = currentValue.Type;
+    itemText = currentValue.Text;
+  }
+
+  if (itemType !== 0) return currentValue;
+
+  // Naive approach, needs massive improvement
+  const contextLength = 50;
+  let strContext = '';
+
+  if (diffs[index - 1] && diffs[index - 1][0] !== 0) {
+    strContext += `${itemText.substring(0, contextLength)}\n`;
+  }
+  if (diffs[index + 1] && diffs[index + 1][0] !== 0) {
+    strContext += `\n${itemText.substring(itemText.length - contextLength)}`
+  }
```

Comment:
You need to make sure the two ends of the context here don’t overlap. For example:

```
[
  [1, "hello "],
  [0, "there "],
  [-1, "Kevin"]
]
```

…would double up the unchanged item in the middle into `"there there "` because it’s so short.

It also might be worth considering whether to put an ellipsis or some other marker in place of text that is removed from the unchanged section.

1. The simple way to do that would be to simply replace that part of the string with the ellipsis (or whatever)
2. The extra fancy way might be to split this no-change entry in two, then insert some kind of placeholder in the middle that `DiffItem` knows how to render as a separator. This gives you a lot more visual flexibility. E.g. you’d be shooting for output like:

    ```
    [
      [1, "some added text"],
      [0, "the part of the unchanged context near the start"],
      [100, "SEPARATOR"],
      [0, "the part of the unchanged context near the end"]
      [1, "more added text"]
    ]
    ```

    …and change `DiffItem` to render something special for `100`-type items.